adding more features

jerrypena1 opened the pull request:
Adding a save button onto the main header component so the admin can save the current tasks directly to supabase/admin. Adding an import button to the admin so that you don't have to start with a brand new tasks list every time.

jerrypena1 committed:
adding more features

jerrypena1 committed:
Update vite.config.ts

jerrypena1 committed:
Update App.tsx

jerrypena1 committed:
Update App.tsx

jerrypena1 committed:
adding even more functionality

leex279 commented on src/App.tsx:
```diff
@@ -132,13 +132,14 @@ export default function App() {
       </div>
       {error && <ErrorNotification message={error} onClose={() => setError(null)} />}
 
-      <div className="max-w-2xl mx-auto px-4 py-12 sm:px-6 lg:px-8">
```
Please leave it with the original one. I dont think its a good idea to put it full with, cause it makes it hard to read. I´ve got another PR from myself to extend it by 25% and other stuff like deeplinking for the example lists.

jerrypena1 replied:
will do 👍

leex279 commented:
Thanks for the PR, very nice. Please change the one thing I mentioned above and I will merge the rest. Great work :)

jerrypena1 committed:
restoring original main display classes

jerrypena1 commented:
I've made the change you requested.

jerrypena1 committed:
Update CodeBlockEditor.tsx

There was an error because passing in the language was only partially implemented

jerrypena1 committed:
Update useAuth.ts

Revised the setIsAdmin logic

jerrypena1 committed:
Update TaskEditForm.tsx

add language to CodeBlockEditor implementation in TaskEditForm to match TaskInput

jerrypena1 commented:
@leex279 I've made a few minor changes that I noticed I had to fix when I first forked the project. I had to do it again when I was testing the deeplinking branch, so I thought I'd just add them here (adding missing language props to the codeeditor and updating the setIsAdmin logic).

leex279 commented:
@jerrypena1 thx, I did a few more changes and pushed the deeplinking to stable already.
I want to merge this now into main, so we can test it there properly.

Can you resolve the merge conflicts pls :)

jerrypena1 committed:
Update TaskInput.tsx

jerrypena1 committed:
Update App.tsx

fixing merge errors

jerrypena1 committed:
Update useAuth.ts

fix merge errors

jerrypena1 committed:
Merge branch 'stable' into more-features

jerrypena1 commented:
@leex279 Merge conflicts resolved.

leex279 commented:
@jerrypena1 thanks. merged to main and stable.

**Summary of your changes:**

- Added the ability for admins to save task lists via a new "Save List" button in the header.
- Introduced SaveModal and SaveImportModal components for naming and saving lists, with an option to save as example lists.
- Admin Dashboard now supports importing task lists from JSON files and saving imported tasks as new lists.
- ListEditor component now supports importing tasks from JSON files directly into a list being edited.
- Improved error handling and feedback when saving lists or importing fails.
- Updated CodeBlockEditor and related components to properly pass language and code values.
- Fixed minor bugs and improved function signatures in components related to task editing and input.
- UI enhancements: added visual feedback for saving, improved modal dialogs, and streamlined task import/export experience for admins.
